test/distributed/test_c10d.py

```diff
@@ -4218,100 +4218,121 @@ def forward(self, inp):
             x = checkpoint(self.l2, x)
             return x
 
-    def _test_ddp_checkpointing(self, checkpoint_once, process_group, use_bucket_view, find_unused_parameters=False):
+    def _prepare_dummy_data(self):
+        ddp_bs = 16
+        bs = ddp_bs * self.world_size
+        input = torch.rand((bs, 20), device="cuda", requires_grad=True)
+        target = torch.randn((bs, 20), device="cuda")
```

Comment:
Maybe use `torch.cuda.current_device()` to ensure the tensors are on the right device?

Reply:
torch.cuda.set_device(self.rank) was set at the beginning